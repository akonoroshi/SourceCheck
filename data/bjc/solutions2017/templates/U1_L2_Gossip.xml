<project name="U1_L2_Gossip" app="iSnap 2.1, http://go.ncsu.edu/isnap" version="1"><data>{}</data><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAAD10lEQVR4Xu3aPWuTARTF8ZvmKRiILS1FOyj5EIEsFotLVwvikMmh4N6IDurmEHASOgqtggod+gUcuttvIBkFF8Gl1AZt8kgyaH0ZMuXk3P4DHQoPufee84O2NJWyLMvgRQKiBCoAFCXP2HECAASCNAEASuNnOAAxIE0AgNL4GQ5ADEgTAKA0foYDEAPSBAAojZ/hAMSANAEASuNnOAAxIE0AgNL4GQ5ADEgTAKA0foYDEAPSBAAojZ/hAMSANAEASuNnOAAxIE0AgNL4GQ5ADEgTAKA0foYDEAPSBAAojZ/hAMSANAEASuNnOAAxIE0AgNL4GQ5ADEgTAKA0foYDEAPSBAAojZ/hAMSANAEASuNnOAAxIE0AgNL4GQ5ADEgTAKA0foYDEAPSBAAojZ/hAMSANAEASuNnOAAxIE0AgNL4GQ5ADEgTAKA0foYDEAPSBAAojZ/hAMSANAEASuNnOAAxIE0AgNL4GQ5ADEgTAKA0foYDEAPSBAAojZ/hAMSANAEASuNnOAAxIE0AgNL4GW4JsNlsxv7+fjQajZifn6dF4wQsAY7y3tzcjEqlMv4qyzL29vZicXFx/D0vnwRsAR4dHUW32/0n6bOzs2i1WrG9vR31et2niQu6qS3Aw8PD2NnZ+aO24XAY1Wo11tbWotPpXNBKvc62BNjv96Pdbo9/9I5+B6zVarG7uxtFUXilz7ZhCXB9fT0ODg5iZWWFCs0TsARonjnrn0sAgHCQJgBAafwMByAGpAkAUBo/wwE44wYGg0F8+34cdx7fiOXOx3i59DUu1xdmfOvJ1wPg5FlN7cmT0+N48vxBfL79Luaunvya++LSp1hduja1PaYxCIDTSHnCGRuvF2J54/i/TzffP42H955N+E4+jwFwBru6++hWFPc/xLB2GpW53wu+ufIjimqu//YAcAYBnl+p1+vF1qubsbr1JYqiGm+v91N94geAMw7w7/VGf5SMPnCR5QXALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3gFA0+KyrA3ALE2a3vETswEtt7iQN1oAAAAASUVORK5CYII=</thumbnail><stage name="Stage" guid="27109659-f2e2-4b9c-9a46-e46e1ee2b009" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" codify="false" inheritance="false" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Sprite(2)" idx="2" x="56.069796390999954" y="-46.8760690570914" heading="-90" scale="1" rotation="1" draggable="true" costume="0" color="87.20999999999998,229.5,0" pen="tip" id="8"><costumes><list id="9"></list></costumes><sounds><list id="10"></list></sounds><variables></variables><blocks></blocks><scripts><script x="16.666666666666657" y="16.666666666666657"><custom-block id="44" s="who2"><comment w="308.8333333333333" collapsed="true">A.  Click this new block several times to see what it does.</comment></custom-block></script><script x="16.666666666666657" y="47"><custom-block id="45" s="gossip2"><comment w="219.33333333333337" collapsed="true">B.  Click several times to see what it does.</comment></custom-block></script><script x="16.666666666666657" y="77.33333333333331"><block id="46" s="receiveMessage"><l>your turn</l><comment w="265.3333333333333" collapsed="true">C. Click this script and watch it run several times. </comment></block><block id="47" s="doWait"><l>1</l></block><block id="48" s="bubble"><custom-block id="49" s="gossip2"></custom-block></block></script></scripts></sprite><sprite name="Sprite" idx="1" x="-75.93020360899868" y="-14.876069057091115" heading="90" scale="1" rotation="1" draggable="true" costume="0" color="80,80,80" pen="tip" id="23"><costumes><list id="24"></list></costumes><sounds><list id="25"></list></sounds><variables></variables><blocks></blocks><scripts><script x="16.666666666666657" y="13.333333333333343"><custom-block id="50" s="who"><comment w="308" collapsed="true">A. Click this red block several times to see what this does.</comment></custom-block></script><script x="16.666666666666657" y="43.666666666666686"><custom-block id="51" s="does what"><comment w="271" collapsed="true">B.  Click this block several times to see what it does.</comment></custom-block></script><script x="16.666666666666657" y="74"><custom-block id="52" s="gossip"><comment w="310.3333333333333" collapsed="true">C.  Click this green block several times to see what it does.</comment></custom-block></script><script x="53.33333333333337" y="150"><block id="1126" s="receiveInteraction"><l><option>clicked</option></l></block><block id="975" s="setScale"><l>100</l></block><block id="54" s="bubble"><custom-block id="55" s="gossip"></custom-block><comment w="313.6666666666667" collapsed="false">D.  Click this script and watch the sprites talk several times.&#xD;Wait until both sprites talk before clicking it again!</comment></block><block id="56" s="doBroadcast"><l>your turn</l></block></script></scripts></sprite></sprites></stage><hidden></hidden><headers></headers><code></code><blocks><block-definition s="who" type="reporter" category="lists" guid="5e9eee88-7c91-4930-bc99-f0eb3fe5b661" isImported="false"><comment x="0" y="0" w="425.33333333333326" collapsed="true">Change some list items. To add more items, click the right-arrow of the list block.</comment><header></header><code></code><inputs></inputs><script><block id="25" s="doReport"><block id="26" s="reportListItem"><l><option>any</option></l><block id="27" s="reportNewList"><list><l>Señora</l><l>Ms. C</l><l>my cat</l><l>Hannah</l><l>Jake</l></list></block></block></block></script></block-definition><block-definition s="does what" type="reporter" category="lists" guid="387bb696-7daa-4610-bd6e-2f387162cb92" isImported="false"><comment x="0" y="0" w="425.33333333333326" collapsed="true">Change some list items. To add more items, click the right-arrow of the list block.</comment><header></header><code></code><inputs></inputs><script><block id="28" s="doReport"><block id="29" s="reportListItem"><l><option>any</option></l><block id="30" s="reportNewList"><list><l>listened to</l><l>loves</l><l>ran away from</l><l>helped</l><l>is in a band with</l></list></block></block></block></script></block-definition><block-definition s="gossip" type="reporter" category="operators" guid="6b3826fa-344e-45a0-b183-20059e64b2f4" isImported="false"><header></header><code></code><inputs></inputs><script><block id="31" s="doReport"><block id="32" s="reportJoinWords"><list><custom-block id="33" s="who"></custom-block><l> </l><custom-block id="34" s="does what"></custom-block><l> </l><custom-block id="35" s="who"></custom-block></list><comment w="254.00000000000003" collapsed="false">JOIN puts the sentence together. &#xD;The two dots between the phrases stand for spaces.</comment></block></block></script></block-definition><block-definition s="who2" type="reporter" category="lists" guid="6e3bc340-3e9b-4943-b2d4-91964d54dc64" isImported="false"><comment x="0" y="0" w="425.33333333333326" collapsed="true">Change some list items. To add more items, click the right-arrow of the list block.</comment><header></header><code></code><inputs></inputs><script><block id="36" s="doReport"><block id="37" s="reportListItem"><l><option>any</option></l><block id="38" s="reportNewList"><list><l>Gabrielle</l><l>Malik</l><l>our neighbor&apos;s dog</l><l>Mr. C</l><l>nobody</l><l>nobody</l></list></block></block></block></script></block-definition><block-definition s="gossip2" type="reporter" category="operators" guid="3282198d-d0b3-4fc2-9ff2-784660ffbcda" isImported="false"><header></header><code></code><inputs></inputs><script><block id="39" s="doReport"><block id="40" s="reportJoinWords"><list><custom-block id="1056" s="gossip response"></custom-block><l> </l><custom-block id="41" s="who2"></custom-block><l> </l><custom-block id="42" s="does what"></custom-block><l> </l><custom-block id="43" s="who"></custom-block></list></block><comment w="254.00000000000003" collapsed="false">JOIN puts the sentence together. &#xD;The two dots between the phrases are spaces.</comment></block></script></block-definition><block-definition s="gossip response" type="reporter" category="lists" guid="49854376-7190-421c-ba10-dc33614e5bbe" isImported="false"><header></header><code></code><inputs></inputs><script><block id="983" s="doReport"><block id="992" s="reportListItem"><l><option>any</option></l><block id="996" s="reportNewList"><list><l>hi</l><l>what</l><l>sup</l></list></block></block></block></script></block-definition></blocks><variables></variables><editing></editing></project>