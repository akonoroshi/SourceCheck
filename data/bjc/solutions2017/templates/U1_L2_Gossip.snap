snapshot {
    stage {
        sprite {
            @repeat (3, 3) {
                script {
                    evaluateCustomBlock
                }
            }
            script {
                @or {
                    receiveInteraction,
                    receiveGo
                }
                @optional { setScale }
                bubble(evaluateCustomBlock)
                doBroadcast
            }
        }
        sprite {
            @repeat (2, 2) {
                script {
                    evaluateCustomBlock
                }
            }
            @optional {
                script {
                    receiveGo
                    setScale
                }
            }
            script {
                receiveMessage
                doWait
                bubble(evaluateCustomBlock)
            }
        }
    }
    @defBlock LIST_BLOCK {
        customBlock {
            script {
                doReport(reportListItem(literal, reportNewList(list(
                    @anything
                ))))
            }
        }
    }
    @block LIST_BLOCK
    @block LIST_BLOCK
    customBlock {
        script {
            doReport(reportJoinWords(list(evaluateCustomBlock, literal, evaluateCustomBlock, literal, evaluateCustomBlock)))
        }
    }
    @block LIST_BLOCK

    @defVar (RESP) {
        customBlock {
            script {
                doReport(reportJoinWords(list(
                    @if (RESP) {
                        evaluateCustomBlock
                    }
                    @optional { literal }
                    evaluateCustomBlock, literal, evaluateCustomBlock, literal, evaluateCustomBlock
                )))

            }
        }
        @if (RESP) {
            @block LIST_BLOCK
        }
    }
}
